File: 2/list-2.py
class Carro:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidade = 0  
    
    def acelerar(self, valor):
        self.velocidade += valor
        print(f"O carro acelerou. Velocidade atual: {self.velocidade} km/h")
    
    def frear(self, valor):
        self.velocidade -= valor
        if self.velocidade < 0:
            self.velocidade = 0
        print(f"O carro freou. Velocidade atual: {self.velocidade} km/h")
    
    def exibir_velocidade(self):
        print(f"Velocidade atual: {self.velocidade} km/h")

def main():
    meu_carro = Carro("Toyota", "Corolla")
    meu_carro.acelerar(30)
    meu_carro.exibir_velocidade()
    meu_carro.frear(10)
    meu_carro.exibir_velocidade()
    meu_carro.frear(25)
    meu_carro.exibir_velocidade()

if __name__ == "__main__":
    main()

File: 2/list-2.go
package main

import "fmt"

type Carro struct {
	Marca      string
	Modelo     string
	Ano        int
	Velocidade int
}

func (c *Carro) Acelerar() {
	c.Velocidade += 10
}

func (c *Carro) Frear() {
	if c.Velocidade >= 10 {
		c.Velocidade -= 10
	}
}

func (c *Carro) ExibirVelocidade() string {
	return fmt.Sprintf("Velocidade atual: %d km/h", c.Velocidade)
}

func main() {
	carro := Carro{
		Marca:      "Toyota",
		Modelo:     "Corolla",
		Ano:        2020,
		Velocidade: 0,
	}

	carro.Acelerar()
	fmt.Println(carro.ExibirVelocidade())

	carro.Frear()
	fmt.Println(carro.ExibirVelocidade())
}
